src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Parser.cs

```diff
@@ -0,0 +1,88 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.AspNetCore.Http.SourceGeneration.StaticRouteHandlerModel;
+
+internal static class StaticRouteHandlerModelParser
+{
+    public static EndpointRoute GetEndpointRouteFromArgument(IArgumentOperation argumentOperation)
+    {
+        var syntax = argumentOperation.Syntax as ArgumentSyntax;
+        var expression = syntax.Expression as LiteralExpressionSyntax;
+        return new EndpointRoute
+        {
+            RoutePattern = expression.Token.ValueText
+        };
+    }
+
+    public static EndpointResponse GetEndpointResponseFromMethod(IMethodSymbol method)
+    {
+        return new EndpointResponse
+        {
+            ContentType = "plain/text",
+            ResponseType = method.ReturnType.ToString(),
+        };
+    }
+
+    public static Endpoint GetEndpointFromOperation(IInvocationOperation operation)
+    {
+        var routePatternArgument = operation.Arguments[1];
+        var method = ResolveMethodFromOperation(operation.Arguments[2]);
```

Comment:
Does this work when arguments are not used in the same parameter order (when using named arguments)?